pkg/services/datasources/service/datasource.go

```diff
@@ -534,21 +534,23 @@ func (s *Service) UpdateDataSource(ctx context.Context, cmd *datasources.UpdateD
 			}
 		}
 
-		// TODO: we will eventually remove this check for moving the resource to it's separate API
-		if s.features != nil && s.features.IsEnabled(ctx, featuremgmt.FlagTeamHttpHeaders) && !cmd.OnlyUpdateLBACRulesFromAPI {
-			s.logger.Debug("Overriding LBAC rules with stored ones",
-				"reason", "update_lbac_rules_from_datasource_api",
-				"action", "use_updateLBACRules_API",
+		// preserve existing lbac rules when updating datasource if we're not updating lbac rules
+		// TODO: Refactor to store lbac rules separate from a datasource
+		allowLBACRuleUpdates := cmd.AllowLBACRuleUpdates && s.features != nil && s.features.IsEnabled(ctx, featuremgmt.FlagTeamHttpHeaders)
+		if !allowLBACRuleUpdates {
```

Comment:
Small bug, you need the feature flag enabled to care about the flag.
```suggestion
		if s.features != nil && s.features.IsEnabled(ctx, featuremgmt.FlagTeamHttpHeaders) && !allowLBACRuleUpdates {
```